agent/handlers/v2/response.go

```diff
@@ -99,16 +103,23 @@ func NewTaskResponse(
 	}
 
 	if propagateTags {
-		propagateTagsToMetadata(ecsClient, containerInstanceArn, taskARN, resp, includeV4Metadata)
+		propagateTagsToMetadata(endpointID, ecsClient, state, containerInstanceArn, taskARN, resp, includeV4Metadata)
 	}
 
 	return resp, nil
 }
 
 // propagateTagsToMetadata retrieves container instance and task tags from ECS
-func propagateTagsToMetadata(ecsClient ecs.ECSClient, containerInstanceARN, taskARN string, resp *tmdsv2.TaskResponse, includeV4Metadata bool) {
-	containerInstanceTags, err := ecsClient.GetResourceTags(containerInstanceARN)
+func propagateTagsToMetadata(endpointID string, ecsClient ecs.ECSClient, state dockerstate.TaskEngineState, containerInstanceARN, taskARN string,
+	resp *tmdsv2.TaskResponse, includeV4Metadata bool) {
+	// Extract the container name and create a context for CloudTrail visibility
+	ctx := context.Background()
+	if containerName, found := state.ContainerNameByV3EndpointID(endpointID); found {
```

Comment:
Since this function allows passing empty string as `endpointID`, should it also look up container name only when `endpointID` is not empty? It would be also be good to document that the function allows `endpointID` to be empty.

Reply:
Updated